edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentController.cs

```diff
@@ -78,22 +78,22 @@ public async Task<EdgeDeploymentStatus> DeployModulesAsync(ModuleSet desiredModu
                 };
 
                 var desiredServices = desiredModules.Modules
-                    .Select(module => this.serviceMapper.CreateService(moduleIdentities[module.Key], module.Value as KubernetesModule, labels[module.Key]))
+                    .Select(module => this.serviceMapper.CreateService(moduleIdentities[module.Key], (KubernetesModule)module.Value, labels[module.Key]))
```

Comment:
Why move from as to cast? Cast will throw an exception - is that what you want? As allows us to handle the error w/o an excpetion by checking against null.

Reply:
I don't have a major opinion on this item yet - maybe @dmolokanov does. The code downstream is going to NullException anyway, should 'as' operator assign null. I wanted to failfast sooner here.

Reply:
Yes, if a cast is failed here - it means that we failed to maintain only KubernetesModule here on a call stack before this place. It's better to fail fast here :)